sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -206,9 +207,12 @@ def _upload_entity_df_into_bigquery(project, dataset_name, entity_df, client) ->
     # First create the BigQuery dataset if it doesn't exist
     dataset = bigquery.Dataset(f"{client.project}.{dataset_name}")
     dataset.location = "US"
-    client.create_dataset(
-        dataset, exists_ok=True
-    )  # TODO: Consider moving this to apply or BigQueryOfflineStore
+
+    try:
+        client.get_dataset(dataset)
+    except NotFound:
+        # Only create the dataset if it does not exist
+        client.create_dataset(dataset)
```

Comment:
Is there a reason we removed `exists_ok=True`? Wouldn't we introduce a race condition here?

Reply:
I decided to remove it because the except block would only be executed if the dataset doesn't exist. I didn't think about race conditions. I'm adding this back.